compiler/rustc_ast/src/ast.rs

```diff
@@ -1931,7 +1931,7 @@ pub enum TyKind {
 
 impl TyKind {
     pub fn is_implicit_self(&self) -> bool {
-        if let TyKind::ImplicitSelf = *self { true } else { false }
+        matches!(*self, TyKind::ImplicitSelf)
```

Comment:
```suggestion
        matches!(self, TyKind::ImplicitSelf)
```

We should also change `TyKind` to be `Copy`... will just throw a PR up for that :shrug: